plugins/obs-qsv11/common_utils_linux.cpp

```diff
@@ -116,16 +311,27 @@ mfxStatus Initialize(mfxVersion ver, mfxSession *pSession,
 		cfg, (const mfxU8 *)"mfxImplDescription.AccelerationMode",
 		impl);
 
+	const char *device_path = NULL;
 	int fd = -1;
-	if (codec == QSV_CODEC_AVC && default_h264_device)
-		fd = open(default_h264_device, O_RDWR);
-	if (codec == QSV_CODEC_HEVC && default_hevc_device)
-		fd = open(default_hevc_device, O_RDWR);
-	if (codec == QSV_CODEC_AV1 && default_av1_device)
-		fd = open(default_av1_device, O_RDWR);
+	if (pmfxAllocator) {
+		obs_video_info ovi;
+		obs_get_video_info(&ovi);
+		struct get_drm_device_params params = {&device_path,
+						       (uint32_t)ovi.adapter};
+		obs_enter_graphics();
+		gs_enum_adapters(get_drm_device, &params);
+		obs_leave_graphics();
+	} else {
+		if (codec == QSV_CODEC_AVC && default_h264_device)
+			device_path = default_h264_device;
+		if (codec == QSV_CODEC_HEVC && default_hevc_device)
+			device_path = default_hevc_device;
+		if (codec == QSV_CODEC_AV1 && default_av1_device)
+			device_path = default_av1_device;
```

Comment:
Aren't these mutually exclusive? Could this be an `if-else if-else if` so the other conditions are not evaluated once one is matched?

Reply:
I generally prefer to leave it if they are mutually exclusive, but I added elses.